lib/logstash/inputs/s3.rb

```diff
@@ -391,12 +400,22 @@ def process_log(queue, key)
     filename = File.join(temporary_directory, File.basename(key))
     if download_remote_file(object, filename)
       if process_local_log(queue, filename, object)
-        lastmod = object.last_modified
-        backup_to_bucket(object)
-        backup_to_dir(filename)
-        delete_file_from_bucket(object)
+        begin
+          # Remote object related operations will fail if it is (re)moved
+          # from the bucket (e.g processed by another instance of Logstash).
+          lastmod = object.last_modified
+          backup_to_bucket(object)
```

Comment:
leads to `Exception: Aws::S3::Errors::NoSuchKey` if the file is processed by another instance